test-utils/src/rpc/client.rs

```diff
@@ -0,0 +1,275 @@
+//! HTTP RPC client.
+
+//---------------------------------------------------------------------------------------------------- Use
+use std::sync::Arc;
+
+use serde::Deserialize;
+use serde_json::json;
+use tokio::task::spawn_blocking;
+
+use monero_serai::{
+    block::Block,
+    rpc::{HttpRpc, Rpc},
+};
+
+use cuprate_types::{TransactionVerificationData, VerifiedBlockInformation};
+
+use crate::rpc::constants::LOCALHOST_RPC_URL;
+
+//---------------------------------------------------------------------------------------------------- HttpRpcClient
+/// An HTTP RPC client for Monero.
+pub struct HttpRpcClient {
+    address: String,
+    rpc: Rpc<HttpRpc>,
+}
+
+impl HttpRpcClient {
+    /// Create an [`HttpRpcClient`].
+    ///
+    /// `address` should be an HTTP URL pointing to a `monerod`.
+    ///
+    /// If `None` is provided the default is used: [`LOCALHOST_RPC_URL`].
+    ///
+    /// Note that for [`Self::get_verified_block_information`] to work, the `monerod`
+    /// must be in unrestricted mode such that some fields (e.g. `pow_hash`) appear
+    /// in the JSON response.
+    ///
+    /// # Panics
+    /// This panics if the `address` is invalid or a connection could not be made.
+    pub async fn new(address: Option<String>) -> Self {
+        let address = address.unwrap_or_else(|| LOCALHOST_RPC_URL.to_string());
+
+        Self {
+            rpc: HttpRpc::new(address.clone()).await.unwrap(),
+            address,
+        }
+    }
+
+    /// The address used for this [`HttpRpcClient`].
+    #[allow(dead_code)]
+    const fn address(&self) -> &String {
+        &self.address
+    }
+
+    /// Access to the inner RPC client for other usage.
+    #[allow(dead_code)]
+    const fn rpc(&self) -> &Rpc<HttpRpc> {
+        &self.rpc
+    }
+
+    /// Request data and map the response to a [`VerifiedBlockInformation`].
+    ///
+    /// # Panics
+    /// This function will panic at any error point, e.g.,
+    /// if the node cannot be connected to, if deserialization fails, etc.
+    pub async fn get_verified_block_information(&self, height: u64) -> VerifiedBlockInformation {
+        #[derive(Debug, Deserialize)]
+        struct Result {
+            blob: String,
+            block_header: BlockHeader,
+        }
+
+        #[derive(Debug, Deserialize)]
+        struct BlockHeader {
+            block_weight: usize,
+            long_term_weight: usize,
+            cumulative_difficulty: u128,
+            hash: String,
+            height: u64,
+            pow_hash: String,
+            reward: u64, // generated_coins
```

Comment:
Is this correct? `reward == generated_coins`? `reward` is the field name in the RPC `get_block` response.

Reply:
no, reward include tx fees, generated coins is without fees